### PR TITLE
ENH: correct bounds warnings in `minimize`

### PR DESCRIPTION
**Pull request by MathieuDutSik**
Some methods use bounds for optimizing and some do not.
Some warnings were given but they were inconsistent.
The PR makes a uniform warning about it that is coherent
over all methods.

### CODE REVIEW

**Review by tupui — COMMENTED** [review]
Thank you for the PR @MathieuDutSik.

This is kind of a duplicate of the above check L564. I wonder if we could not merge both. While doing so, we could also consider `_custom` (this is the observed failure, now that you fixed the list, it needs `_custom`). The case is handled bellow but should not raise a warning (BTW there is no stacklevel for all these warnings). What do you think @andyfaff?

Note: we should really make a table in the doc.

File: scipy/optimize/_minimize.py [hunk 1]
```diff
@@ -568,7 +568,8 @@ def minimize(fun, x0, args=(), method=None, jac=None, hess=None,
     if meth in ('nelder-mead', 'l-bfgs-b', 'tnc', 'powell') and np.any(constraints):
         warn('Method %s cannot handle constraints.' % method,
              RuntimeWarning)
-    if meth == 'cobyla' and bounds is not None:
+    if meth not in ('nelder-mead', 'powell', 'l-bfgs-b', 'tnc', 'slsqp',
+                    'trust-constr') and bounds is not None:
```
Comment (tupui) [inline scipy/optimize/_minimize.py:572]:
```suggestion
                    'trust-constr', '_custom') and bounds is not None:
```

Comment (tupui) [reply scipy/optimize/_minimize.py]: mmmm this PR should not have to do anything with that. Could you merge `main` into your PR?

**Comment by mdhaber** [comment]
This is an improvement, so I'll go ahead and merge it. As @tupui mentioned, it would be good to add a table to the docs, and it would be good to add `stacklevel` to these warnings. Furthermore, it would be nice to separate out these tuples of methods that accept different options into `set`s that can be imported into `test_minimize` (e.g. `methods_accept_hessp`) and tested. While we're at it, let's convert to f-strings.

**Comment by MathieuDutSik** [comment]
I did merge the main and the problem persists.

**Pull request merged**